src/test/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilderTest.java

```diff
@@ -0,0 +1,304 @@
+package bio.terra.service.snapshotbuilder.utils;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.equalToCompressingWhiteSpace;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import bio.terra.common.category.Unit;
+import bio.terra.common.exception.BadRequestException;
+import bio.terra.grammar.azure.SynapseVisitor;
+import bio.terra.model.SnapshotBuilderCriteria;
+import bio.terra.model.SnapshotBuilderCriteriaGroup;
+import bio.terra.model.SnapshotBuilderDomainCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataListCriteria;
+import bio.terra.model.SnapshotBuilderProgramDataRangeCriteria;
+import bio.terra.service.snapshotbuilder.query.FilterVariable;
+import bio.terra.service.snapshotbuilder.query.Query;
+import bio.terra.service.snapshotbuilder.query.TablePointer;
+import bio.terra.service.snapshotbuilder.query.TableVariable;
+import java.math.BigDecimal;
+import java.util.List;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+@Tag(Unit.TAG)
+class CriteriaQueryBuilderTest {
+
+  @Test
+  void generateRangeCriteriaFilterProducesCorrectSql() {
```

Comment:
There's probably a mix of conventions in the TDR code base. My preferred naming is to name the test after the method being tested, so this would be
```suggestion
  void generateFilterForRangeCriteria() {
```

I think in general shorter names are better (easier to read) so I'm always looking for ways to make names shorter.